contrib/ceph-build-config.sh

```diff
@@ -0,0 +1,530 @@
+# version '1.0.0' might also match a version '11.0.0' unless the search is for 'v1.0.0'.
+function convert_version_to_version_tag () {
+  version="${1}"
+  version_minus_build_number="${version%-*}"
+  echo "v${version_minus_build_number}"
+}
+
+
+# We use the 8-char date (YYYYMMDD) as the build number to simplify things
+function generate_new_build_number () {
+  date --utc +%Y%m%d
+}
+
+# Given an image repo, version tag, and buidl number: construct the full tag to push
+function construct_full_push_image_tag () {
+  version_tag="${1}" ; image_repo="${2}" ; build_number="${3}"
+  if [ -z "${build_number}" ]; then
+    # If build number is empty string, don't append it
+    echo "${PUSH_LIBRARY}/${image_repo}:${version_tag}"
+    return
+  fi
+  echo "${PUSH_LIBRARY}/${image_repo}:${version_tag}-${build_number}"
+}
+
+# Given a full tag, extract the build number from it
+function extract_full_tag_build_number () {
+  full_tag="${1}"
+  echo "${full_tag##*-}"
+}
+
+# Return the most recent build number between 2 build numbers
+function latest_build_number () {
+  printf '%s\n%s' "${1}" "${2}" | sort --version-sort | tail -1
+}
+
+# Return the version string with the build number stripped from the end
+# Build number is assumed to be <version>-<build-num>,; the hyphen and build-num are both stripped
+# build-num itself is assumed not to contain hyphens.
+function convert_version_to_version_without_build () {
+  version="${1}"
+  echo "${version%-*}"
+}
+
+# Convert a version tag to a major-minor-version tag (e.g., v12.2.2 -> v12.2)
+function convert_version_tag_to_major_minor_tag () {
+  version_tag="${1}"
+  echo "${version_tag%.*}"
+}
+
+# Convert a version tag to a major-version tag (e.g., v12.2.2 -> v12)
+function convert_version_tag_to_major_tag () {
+  version_tag="${1}"
+  echo "${version_tag%%.*}"
+}
+
+# Given a version tag, return the minor version number
+function extract_minor_version () {
+  version_tag="${1}"
+  major_and_minor="${version_tag%.*}"
+  echo "${major_and_minor#*.}"
+}
+
+
+#===================================================================================================
+# Image repository querying
+#===================================================================================================
+
+# Return the full tags matching a given version tag on the repository
+function get_tags_matching () {
+  version_tag="${1}" ; repository="${2}"
+  # DockerHub API limits page_size to 100, so we must loop through the pages
+  # It would be super cool if the DockerHub HTTP API had a filter option ...
+  tag_list_url="${PUSH_REGISTRY_URL}/v2/repositories/${PUSH_LIBRARY}/${repository}/tags/?page_size=100"
+  all_matching_tags=''
+  page=1
+  while response="$(curl --silent --fail --list-only --location \
+                      "${tag_list_url}&page=${page}")"; do
+    matching_tags="$(echo "${response}" | \
+              jq -r ".results[] | select(.name | match(\"${version_tag}\")) | .name")" || exit $?
```

Comment:
I dunno, but it would be good if it could be installed. Otherwise, the json needs to be parsed with some sort of manual solution, which will probably be fragile.